test/ref_ops_test.cpp

```diff
@@ -1330,6 +1339,56 @@ TEST_CASE(div_test)
     EXPECT(migraphx::verify_range(results_vector, gold));
 }
 
+TEST_CASE(div_zero_compile_trap_after_passes)
+{
+    migraphx::program p;
+    auto* mm  = p.get_main_module();
+    auto zero = mm->add_literal(0.0f);
+    auto x    = mm->add_parameter("x", {migraphx::shape::float_type, {1}});
+    mm->add_instruction(migraphx::make_op("div"), x, zero);
+    run_pass(*mm);
+
+    bool result = false;
+    try
+    {
+        p.compile(migraphx::ref::target{});
+    }
+    catch(const std::runtime_error& e)
+    {
+        (void)e;
+        result = true;
+    }
+    EXPECT(result);
+}
+
+TEST_CASE(div_zero_compile_trap_long_program_after_passes)
+{
+    migraphx::program p;
+    auto* mm  = p.get_main_module();
+    auto zero = mm->add_literal(0.0f);
+    auto two  = mm->add_literal(2.0f);
+    auto x    = mm->add_parameter("x", {migraphx::shape::float_type, {1}});
+    auto y    = mm->add_parameter("y", {migraphx::shape::float_type, {1}});
+    auto div0 = mm->add_instruction(migraphx::make_op("div"), x, zero);
+    auto mul  = mm->add_instruction(migraphx::make_op("mul"), two, div0);
+    auto add  = mm->add_instruction(migraphx::make_op("add"), y, mul);
+    mm->add_instruction(migraphx::make_op("sub"), y, add);
+
+    run_pass(*mm);
+
+    bool result = false;
+    try
+    {
+        p.compile(migraphx::ref::target{});
```

Comment:
use `EXPECT(test_throws())`

Reply:
Tried that and EXPECT_THROW(test_throws()) seems to be breaking builds. What function did you mean?

Reply:
` EXPECT(test::throws([&]{p.compile(migraphx::ref::target{});}));`

Reply:
Similarly in other  places.

Reply:
Done